testgrid/cmd/gatherer/issuescraper.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gatherer
+
+import (
+	"encoding/json"
+	"github.com/sirupsen/logrus"
+	"github.com/tomnomnom/linkheader"
+	"io"
+	"k8s.io/test-infra/ghproxy/ghcache"
+	"net/http"
+)
+
+// JSONPresence detects if a particular key is present without actually parsing any of the data
+type JSONPresence struct {
+	present bool
+}
+
+func (j *JSONPresence) UnmarshalJSON([]byte) error {
+	j.present = true
+	return nil
+}
+
+type GitHubIssueScraper struct {
+	githubRoundTripper http.RoundTripper
+	RepoInfix          string
+}
+
+type GitHubIssue struct {
+	IssueId     int          `json:"number"`
+	Title       string       `json:"title"`
+	CommentsUrl string       `json:"comments_url"`
+	NumComments int          `json:"comments"`
+	Comments    []string     `json:"-"` //ignore
+	PullRequest JSONPresence `json:"pull_request"`
+	Body        string       `json:"body"`
+}
+
+func (g *GitHubIssueScraper) GetIssues() []GitHubIssue {
+
+	thisPage := "https://api.github.com/repos/" + g.RepoInfix + "/issues"
+	issues := make([]GitHubIssue, 0)
+
+	for {
+		resp := g.askGitHub(thisPage)
+
+		links := linkheader.Parse(resp.Header.Get("Link"))
+		issues = append(issues, g.parseIssuePage(resp.Body)...)
+
+		var nextPage string
+		for _, link := range links {
+			if link.Rel == "next" {
+				nextPage = link.URL
+			}
+		}
+
+		logrus.Infof("This Page: %s, Next Page: %s", thisPage, nextPage)
+		if nextPage == "" {
+			break
+		}
+		thisPage = nextPage
+	}
+
+	return issues
+}
+
+func NewIssueScraper(ownerAndRepo string) GitHubIssueScraper {
+	return GitHubIssueScraper{
+		RepoInfix: ownerAndRepo,
+	}
+}
+
+func (g *GitHubIssueScraper) initializeCache() {
+	rt := &http.Transport{}
+	g.githubRoundTripper = ghcache.NewMemCache(rt, 3)
+}
+
+func (g *GitHubIssueScraper) parseCommentsPage(body io.Reader) []string {
```

Comment:
nit: Just parseComments?

Reply:
I think I prefer the "Page" portion to indicate that it's reading a Comments webpage response and not, say, a comments struct.

"Page" may not be the best way to indicate that. I'm thinking "parseCommentsAPI"?

Reply:
"Response" maybe?